test/Microsoft.AspNetCore.Server.Kestrel.FunctionalTests/RequestTests.cs

```diff
@@ -428,59 +429,44 @@ public void CanUpgradeRequestWithConnectionKeepAliveUpgradeHeader()
         [InlineData("http://localhost/", true, "/")]
         [InlineData("https://localhost/", true, "/")]
         [InlineData("http://localhost", true, "")]
+        // bad requests
```

Comment:
Write a separate test method for bad requests. Also, you can add those to the `BadHttpRequestTests` class.